app/policies/application_policy.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# Base policy class
+class ApplicationPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+  end
+
+  def index?
+    false
+  end
+
+  def create?
+    false
+  end
+
+  def update?
+    false
+  end
+
+  def destroy?
+    false
+  end
+
+  # Base Scope class
+  class Scope
+    attr_reader :user, :scope
+
+    def initialize(user, scope)
+      @user = user
+      @scope = scope
+    end
+
+    def resolve
+      scope.all
```

Comment:
We should remove this an define the Scopes on a _per-class basis._

(If we did want to put it here for some reason, then `resolve` should be:
 if the user is an admin, the scope is `all`
 else the scope is none.  (return an empty list)
)

Reply:
I've only retained the constructor in the base scope class.